pkg/appgw/frontend_listeners.go

```diff
@@ -167,10 +167,23 @@ func (c *appGwConfigBuilder) newListener(cbCtx *ConfigBuilderContext, listenerID
 			FrontendIPConfiguration: resourceRef(*frontIPConfiguration.ID),
 			FrontendPort:            resourceRef(*frontendPort.ID),
 			Protocol:                protocol,
-			HostName:                &listenerID.HostName,
+			HostName:                nil,
+            Hostnames:               nil,
 		},
 	}
 
+	// Use only the 'Hostnames' field as application gateway allows either 'HostName' or 'Hostnames'
+	if hostnames, err := listenerID.getHostNames(); err == nil {
+		if hostnames != nil && len(hostnames) > 1 {
+			listener.Hostnames = &hostnames
+		} else {
+			listener.HostName = &hostnames[0]
```

Comment:
It looks like `hostnames` could be `nil` here
Perhaps the `} else {` should be `} else if hostnames != nil {` or even pull the `hostnames==nil` check up into the first conditional

Reply:
Thanks for correcting.  Hostnames cant be nil so will modify accordingly

Reply:
Fixed. Let me know how it looks